src/ru/fizteh/fivt/students/AndrewTimokhin/FileMap/JUnit/Reader.java

```diff
@@ -0,0 +1,101 @@
+package ru.fizteh.fivt.students.AndrewTimokhin.FileMap.JUnit;
+
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Vector;
+
+/**
+ * Класс @class Reader отвечает за физическое чтение данных из файловой системы.
+ * 
+ *
+ * @author Timokhin Andrew
+ */
+
+public class Reader {
+    final int totalSubDir = 16;
+    final String direct = ".dir";
+
+    public void read(TableProviderImplements tp) throws IOException {
+        StringBuilder keyBuilder = new StringBuilder();
+        StringBuilder valueBuilder = new StringBuilder();
+        int length;
+        String path = tp.dir;
+        Vector<TableImplement> tables = new Vector<TableImplement>();
```

Comment:
"Unlike the new collection implementations, Vector is synchronized. If a thread-safe implementation is not needed, it is recommended to use ArrayList in place of Vector."